test/apps/iodemo/ucx_wrapper.cc

```diff
@@ -515,6 +566,48 @@ bool UcxConnection::recv_data(void *buffer, size_t length, uint32_t sn,
     return process_request("ucp_tag_recv_nb", ptr_status, callback);
 }
 
+bool UcxConnection::send_am(const void *meta, size_t meta_length,
+                            const void *buffer, size_t length,
+                            UcxCallback* callback)
+{
+    if (_ep == NULL) {
+        return false;
+    }
+
+    ucp_request_param_t param;
+    param.op_attr_mask = UCP_OP_ATTR_FIELD_CALLBACK  |
+                         UCP_OP_ATTR_FIELD_FLAGS;
+    param.cb.send      = common_request_callback_nbx;
+    param.flags        = UCP_AM_SEND_REPLY;
+    param.datatype     = 0; // make coverity happy
+
+    ucs_status_ptr_t sptr = ucp_am_send_nbx(_ep, AM_MSG_ID, meta, meta_length,
+                                            buffer, length, &param);
+    return process_request("ucp_am_send_nbx", sptr, callback);
+}
+
+bool UcxConnection::recv_am_data(void *buffer, size_t length,
+                                 UcxCallback* callback)
+{
+    if (_ep == NULL) {
+        return false;
+    }
+
+    if (!(_am_desc.recv_param->recv_attr & UCP_AM_RECV_ATTR_FLAG_RNDV)) {
```

Comment:
can we use `UCP_AM_RECV_ATTR_FLAG_DATA` instead?  positive condition easier to read

Reply:
No, because inlined data arrives without `UCP_AM_RECV_ATTR_FLAG_DATA`

Reply:
i see, thanks!